packages/eslint-plugin/src/rules/no-duplicate-type-constituents.ts

```diff
@@ -111,6 +107,13 @@ export default createRule<Options, MessageIds>({
       const cachedTypeMap = new Map<Type, TSESTree.TypeNode>();
       node.types.reduce<TSESTree.TypeNode[]>(
         (uniqueConstituents, constituentNode) => {
+          const constituentNodeType = checker.getTypeAtLocation(
+            parserServices.esTreeNodeToTSNodeMap.get(constituentNode),
+          );
+          if (tsutils.isIntrinsicErrorType(constituentNodeType)) {
+            return [];
+          }
```

Comment:
```suggestion
          if (tsutils.isIntrinsicErrorType(constituentNodeType)) {
            return uniqueConstituents;
          }
```

As I understand this logic, `uniqueConstituents` is a kind of cache for finding duplicates using only AST (since it's faster than calling a checker to make type-level comparisons). See this comment https://github.com/typescript-eslint/typescript-eslint/pull/5728#discussion_r1083520606

If we return an empty array, we discard all previously found unique constituents and fallback to type-level checks, this may result in some performance penalty.

Reply:
Thank you for the guidance. Fixed in 6683e36.